scripts/Diffraction/isis_powder/routines/focus.py

```diff
@@ -48,14 +51,43 @@ def _focus_one_ws(input_workspace, run_number, instrument, perform_vanadium_norm
         summed_empty = common.generate_summed_runs(empty_sample_ws_string=run_details.sample_empty,
                                                    instrument=instrument,
                                                    scale_factor=instrument._inst_settings.sample_empty_scale)
-    if summed_empty is not None:
+
+    # Correct for absorption / multiple scattering if required
+    if paalman_pings and summed_empty and absorb and container_details:
```

Comment:
I wonder if it might be better if we had an overall "absorb" flag that is a boolean and then a second parameter called something like "absorb_method" that is a string and takes values like "simple" and "paalman_pings" (with the default value being "simple"). That way it would be more obvious that setting absorb=false turns off all absorption corrections. What do you think?